### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add embedded dashboards to ClickHouse server. This is a demo project about how to achieve 90% results with 1% effort using ClickHouse features.

Screenshot_20220821_143542
Screenshot_20220821_143631

The dashboard can be self-hosted, examples:
https://pastila.nl/?00399939/ee598c63b98acfe4939898c686b64df5.html
https://pastila.nl/?00399939/04b0fc80fdaaa2dbbc1783f98f4614a6.html

But it is not ready for widespread application and does not compete with any of the full-featured dashboard applications such as Grafana, Superset, Tabler, Retool, Redash, etc.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
@tavplubix this sounds scary: 01108_restart_replicas_rename_deadlock_zookeeper
https://s3.amazonaws.com/clickhouse-test-reports/40461/1aa289f80a963a3ab407edbf636550e22d1acc5d/stateless_tests__address__[2/2].html

File: src/Server/WebUIRequestHandler.h [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <Server/HTTP/HTTPRequestHandler.h>
+#include <re2/re2.h>
```
Comment (serxa) [inline src/Server/WebUIRequestHandler.h:4]: this header is only required in implementation

File: programs/server/dashboard.html [hunk 2]
```diff
@@ -0,0 +1,905 @@
...
+
+/// If it is hosted on server, assume that it is the address of ClickHouse.
+if (location.protocol != 'file:') {
+    host = location.origin;
+    user = 'default';
+}
+
+/// This is just a demo configuration of the dashboard.
+
+let queries = [
```
[the remark this answers is not in the document]
Comment (alexey-milovidov) [reply programs/server/dashboard.html:326]: This PR is the first step, there is a lot to improve in dashboards configuration, but I want to get it merged.

**Review by serxa — APPROVED** [review]
Looks great to me

**Comment by tavplubix** [comment]
> @tavplubix this sounds scary: 01108_restart_replicas_rename_deadlock_zookeeper [https://s3.amazonaws.com/clickhouse-test-reports/40461/1aa289f80a963a3ab407edbf636550e22d1acc5d/stateless_tests__address__[2/2].html](https://s3.amazonaws.com/clickhouse-test-reports/40461/1aa289f80a963a3ab407edbf636550e22d1acc5d/stateless_tests__address__%5B2/2%5D.html)

https://github.com/ClickHouse/ClickHouse/pull/40545 (nothing scary actually, just a race condition)

**Pull request merged**